opm/parser/eclipse/EclipseState/Grid/SatfuncPropertyInitializers.hpp

```diff
@@ -30,31 +30,128 @@ namespace Opm {
 
 namespace Opm { namespace satfunc {
 
+    /// Collection of unscaled/raw saturation range endpoints extracted
+    /// directly from tables of tabulated saturation functions.
     struct RawTableEndPoints
     {
+        /// Connate saturation endpoints
         struct {
+            /// Connate gas saturation.  One value for each saturation region.
+            /// All zero values unless gas is an active phase.
             std::vector<double> gas;
+
+            /// Connate water saturation.  One value for each saturation region.
```

Comment:
I know opinions are split on this; but in my book this extensive commenting is visual noise; in particular the repeated text could be assembled in *one location*

Reply:
Yeah, opinions do differ here.  My point of view is that these kinds of comments reduces the number of times I have to look at function implementations.  As OPM grows&mdash;we're at roughly 1800 source files and 510,000 lines (excluding multiple copies of `pybind11`, but including tests, blanks and comments) of C++ source code now&mdash;it becomes harder and harder to reason about the codebase if we have to constantly look at function bodies.  Also, many tools can show the comments if you mouse over a member or type.

Finally, you should be able to hide the comments in your editor if you don't want to see them.